### DISCUSSION TITLE
Use different access token for specific repository

### DISCUSSION THREAD

**Discussion by fgalaf** [renovatebot/renovate#29838] [Request Help] [answered]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitLab

### Please tell us more about your question or problem

I need to use different Personal Access Tokens for one specific repository which is separate project hosted on GitLab. Currently, my configuration is as follows:
```
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "config:recommended",
    ":dependencyDashboard",
    "group:allNonMajor",
    ":separateMultipleMajorReleases",
    ":automergeDisabled"
  ],
  "hostRules": [
    {
      "matchHost": "gitlab.self-hosted.url",
      "token": "<TOKEN_TO_USE>",
      "hostType": "gitlab-tags"
    }
  ]
}

```
Problem with this config is, that it will use this token for all dependencies that will match our gitlab host URL.
When I enter matchHost value as follows: `https://gitlab.self-hosted.url/path/to/project/repo`, it is not working properly.

My question is, is there a way to only use specific token for single repository? 

Thanks for the reply.

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by rarkins** [comment, marked as answer]
You can, and most likely you have the wrong matchHost. You shouldn't be using the web/html path and instead the API path. You can see the list of URLs attempted in the debug logs (look for the `HTTP statistics` log message towards the end).

**Reply by fgalaf** [reply]
That was the case. Thanks a lot! For anyone having the same issue, correct value was in my case `https://gitlab.self-hosted.url/api/v4/projects/path/to/project/repo`
